[docs][joy-ui] Integrate a count-up feature to the Circular Progress (#38724)
mui/material-ui: docs/data/joy/components/circular-progress/circular-progress.md

{{"demo": "CircularProgressDeterminate.js"}}

> ### Determinate Count Up

Reviewer: ```suggestion

## Common examples

### Count up feature
```

I think we can create a Common examples section for this one ⎯ make sure to move it below the CSS variables playground for consistency with other pages.

> This method uses an external `use-count-up` package, to animate counting up by providing `start, end and duration` values.

Reviewer: ```suggestion
Using the [use-count-up](https://www.npmjs.com/package/use-count-up) package, you can create a counting animation by providing `start`, `end`, and `duration` values.
```

> <codeblock storageKey="package-manager">
> ```bash npm
> npm install use-count-up
> ```
> 
> ```bash yarn
> yarn add use-count-up
> ```
> 
> ```bash pnpm
> pnpm add use-count-up
> ```
> 
> </codeblock>

Reviewer: ```suggestion

```

I'd cut the installation block here and just mention the package link, as suggested below.

{{"demo": "CircularProgressVariables.js", "hideToolbar": true, "bg": "gradient"}}

> ## Common examples
> 
> ### Count up feature

Reviewer 2: ```suggestion
## 3rd-party integration

### use-count-up
```

@danilo-leal What do you think about grouping 3rd-party libraries in `h2`?

We have similar thing on [input](https://mui.com/joy-ui/react-input/#third-party-formatting) but it is `h3`.

Reviewer: I think I'd still leave it in the Common examples, as it would be more consistent with the Input. Maybe the "Third-party integration" title works but it isn't immediately clear when compared to "Third-party formatting" (Input's case). 🤔
